### DISCUSSION TITLE
Why a central package registry?

### DISCUSSION THREAD

**Discussion by mpldr** [vlang/v#21249] [Questions and Answers] [answered]
I was curious about the reasoning behind V using a central package management. After using npm, crates, elm packages, CTAN, CPAN, R packages, and Go modules, I fail to see the deciding benefit of a central registry. Maybe I don't see some positive aspect, so if you could help me understand the move, I would be more than glad.

As I see it now:

## Good

- short(er) package names
- anonymous analytics for package usage
- easier package discovery (though pkg.go.dev arguably solved that issue)

## Bad

- no namespaces
- fosters centralisation
- pad-left like incidents
- less resilient against outages
- possibility to check out experimental branches directly

**Comment by medvednikov** [comment, marked as answer]
You can have it de-centralized with v install https://gihub.com/user/repo
no pad-left like incidents, since there are no top level names (only user.module), and modules can't be deleted.

**Reply by medvednikov** [reply]
Things like `ui` are top level, but this is only for "official" modules.

**Reply by mpldr** [reply]
Ahh thanks, then I misunderstood :)
